### ISSUE TITLE
Version 4.0.6 is no longer compatible with Node 8.6

### ISSUE THREAD

**Issue by willitscale** [micromatch/micromatch#257]
In the recent patch `4.0.6` you've updated the `picomatch` package to `^4.0.2` - https://github.com/micromatch/micromatch/blob/4.0.6/package.json#L41
This is up from patch `4.0.5` where it was `^2.3.1` - https://github.com/micromatch/micromatch/blob/4.0.5/package.json#L41

Picomatch Version `4.0.2` Node Version `>=12` - https://github.com/micromatch/picomatch/blob/4.0.2/package.json#L21
Picomatch Version `2.3.1` Node Version `>=8.6` - https://github.com/micromatch/picomatch/blob/2.3.1/package.json#L19

This is a major change as it updates the required version of node from `8.6` to `12` which is not backwards compatible and breaks SemVer patch rules:

> Patch version Z (x.y.Z | x > 0) MUST be incremented if only backward compatible bug fixes are introduced. A bug fix is defined as an internal change that fixes incorrect behavior.

Source: https://semver.org/

The latest change should have been a major version release and has impacted a number of projects depending on this package such as the grunt-cli which if still used with legacy versions of node will no longer work.

I'd suggest making the current master version `5.0.0` and dropping the `4.0.6` tag.

There's also a number of places in the repo which state this code is compatible with `8.6`:
- https://github.com/micromatch/micromatch/blob/master/README.md?plain=1#L59
- https://github.com/micromatch/micromatch/blob/master/package.json#L34

**Comment by willitscale** [comment]
How do I put up a PR to fix tags?
The owner of the repo would need to run this:
```
git push --delete origin 4.0.6
git checkout master
git tag 5.0.0
git push origin tag 5.0.0
```

**Comment by willitscale** [comment]
Here's a PR to correct the versions in master - https://github.com/micromatch/micromatch/pull/258

**Comment by jonschlinkert** [comment]
fwiw, this is why I don't like being rushed to make patches on these libs. I'm unfortunately traveling until next weekend, and I have very little time to even look at these issues, much less make updates. I'll need some help from the other maintainers if we want to get this fixed before I get back.

**Comment by exussum12** [comment]
https://github.com/micromatch/micromatch/pull/248

appears to be the break

**Comment by Jason3S** [comment]
@jonschlinkert,

> fwiw, this is why I don't like being rushed to make patches on these libs. I'm unfortunately traveling until next weekend, and I have very little time to even look at these issues, much less make updates. I'll need some help from the other maintainers if we want to get this fixed before I get back.

Wow. Who is able to help you out? Is there someone else that can publish to NPM?

Maybe you have a good suggestion on the next steps. This is my take:

1. Make a v4 branch based upon 4.0.5.
2. Bump the version to 4.0.7 on the v4 branch and publish.
3. From master, bump the version to 5.0.0 and publish.

I believe these steps should address the immediate need.

Cleaning up documentation and the rest can happen later.

**Comment by paulmillr** [comment]
fixed in v4.0.7: https://github.com/micromatch/micromatch/tree/v4
